DatadogFlags/Tests/FlagsClientTests.swift

```diff
@@ -38,29 +47,41 @@ final class FlagsClientTests: XCTestCase {
             attributes: ["attr1": "value1", "companyId": "1"]
         )
 
+        // When
         client.setEvaluationContext(context) { result in
             switch result {
             case .success:
-                let boolValue = client.getBooleanValue(key: "boolean-flag", defaultValue: false)
-                let stringValue = client.getStringValue(key: "string-flag", defaultValue: "default")
-                let integerValue = client.getIntegerValue(key: "integer-flag", defaultValue: 0)
-                let doubleValue = client.getDoubleValue(key: "numeric-flag", defaultValue: 0.0)
-                let objectValue = client.getObjectValue(key: "json-flag", defaultValue: [:])
-
-                XCTAssertTrue(boolValue)
-                XCTAssertEqual(stringValue, "red")
-                XCTAssertEqual(integerValue, 42)
-                XCTAssertEqual(doubleValue, 3.14, accuracy: 0.001)
-                XCTAssertEqual(objectValue["key"] as? String, "value")
-                XCTAssertEqual(objectValue["prop"] as? Int, 123)
-
                 expectation.fulfill()
             case .failure(let error):
                 XCTFail("Expected success but got error: \(error)")
             }
         }
 
         waitForExpectations(timeout: 1.0)
+
+        let boolValue = client.getBooleanValue(key: "boolean-flag", defaultValue: false)
+        let stringValue = client.getStringValue(key: "string-flag", defaultValue: "default")
+        let integerValue = client.getIntegerValue(key: "integer-flag", defaultValue: 0)
+        let doubleValue = client.getDoubleValue(key: "numeric-flag", defaultValue: 0.0)
+        let objectValue = client.getObjectValue(key: "json-flag", defaultValue: .null)
+        let typedObjectValue = client.getObjectValue(key: "json-flag", defaultValue: Model(key: "null", prop: 0))
```

Comment:
`AnyValue` is also giving us support for arbitrary `Decodable` types as feature flags values.